File: leetcode/problems/typescript/1732. Find the Highest Altitude/solve.ts
function largestAltitude(gain: number[]): number {
  let maxAlutitude = 0;
  gain.reduce((h, g) => {
    h += g;
    maxAlutitude = Math.max(maxAlutitude, h);
    return h;
  }, 0);
  return maxAlutitude;
}

File: leetcode/problems/cpp/1732. Find the Highest Altitude/solve.cpp
#include <vector>

using namespace std;

class Solution {
public:
  int largestAltitude(vector<int> &gain) {
    int altitude = 0;
    int maxAlutitude = 0;
    for (int i = 0; i < gain.size(); i++) {
      altitude += gain[i];
      maxAlutitude = max(maxAlutitude, altitude);
    }
    return maxAlutitude;
  }
};